docs/manual/src/types/interfaces.md

```diff
@@ -0,0 +1,381 @@
+# Interfaces
+
+Interfaces can have constructors and have methods. In Rust, they are represented as `impl` blocks. In the Kotlin or Swift world, they are a class, so they are often known as "Objects"
```

Comment:
I'd love if the canonical term for this was "object types" in the future. Maybe not this PR. (I think "interfaces" evokes generics too much)

Reply:
I feel this as well, but then the literal part of me wonders: aren't records and enums also "objects"?  Maybe it doesn't matter if "object" invokes the right feeling in users.

I'll also throw out "class" as another term that might work.